Migrate Tree class to Typescript

florianstancioiu opened the pull request:
Migrate `Tree` class to Typescript.

Fixes #65

florianstancioiu committed:
feat: create src directoy

florianstancioiu committed:
feat: add primitive types

florianstancioiu committed:
feat: remove Tree files from root dir

florianstancioiu committed:
test: update Tree import

megheaiulian commented on src/cosmoz-tree.ts:
```diff
@@ -3,7 +3,27 @@ Navigator through object with treelike datastructure and default settings.
 @demo demo/index.html
 */
+
+export type TreeOptions = {
```
Just a preference but i'd name this simply `Options` because we are already in the context of comoz-tree.

florianstancioiu committed:
feat: rename TreeOptions to Options

florianstancioiu committed:
fix: reset lint script to cosmoz-router

florianstancioiu committed:
feat: add Node interface

florianstancioiu committed:
feat: add @types/mocha to prevent linting errors

florianstancioiu committed:
feat: move _sortPathNodes to a separate function

Further work on converting Tree class to typescript (with the help from
Iulian)

florianstancioiu committed:
feat: further work on converting typescript

florianstancioiu committed:
fix: update Tree import in examples script

florianstancioiu committed:
feat: update TODO comment

florianstancioiu committed:
feat: wip - convert tests to typescript

florianstancioiu committed:
feat: update getPathNodes to return only Node[]

florianstancioiu committed:
test: completed tests (with help from Iulian)

florianstancioiu committed:
feat: add dist dir to .eslintignore

florianstancioiu committed:
feat: add test dir to exclude_patterns

megheaiulian commented on package.json:
```diff
@@ -47,6 +49,12 @@
+		"./cosmoz-tree": "./dist/cosmoz-tree.js",
+		"./cosmoz-tree.js": "./dist/cosmoz-tree.js",
+		"./cosmoz-default-tree": "./dist/cosmoz-default-tree.js",
+		"./cosmoz-default-tree.js": "./dist/cosmoz-default-tree.js"
```
Good idea to add both the with and without `.js` export.

cristinecula reviewed (approved):
Nice!
